src/Stack/Storage.hs

```diff
@@ -557,4 +557,6 @@ upgradeChecksSince since = withStorage $ count [UpgradeCheckTimestamp >=. since]
 
 -- | Log in the database that an upgrade check occurred at the given time.
 logUpgradeCheck :: HasConfig env => UTCTime -> RIO env ()
-logUpgradeCheck = withStorage . insert_ . UpgradeCheck
+logUpgradeCheck time = withStorage $ do
+  deleteWhere ([] :: [Filter UpgradeCheck])
```

Comment:
It looks like you'll only ever have a single record in the database, which I agree will work.  What about just updating the existing record if it already exists?

Reply:
It seems like that introduces more complex logic to me: check number of entries, switch between updating and inserting, etc. What's the advantage to updating versus deleting-and-inserting?

Reply:
`upsert` has that logic (although you'd need to add a unique key that you always set to the same value to enforce that there's only one record).  In this case the difference is probably negligible, just avoids the overhead of deleting and creating new records unnecessarily.  In theory, there could be a need for another field in this table someday in the future, in which case deleting the whole record might delete more information than intended.

There may be the case for having one "singleton" table that will only ever contain one record and that we add fields to for any single values that need to be saved, rather than ending up with multiple single-record tables.

Reply:
I've tried out the singleton table approach, sort of. It's a table with two columns: the action and timestamp. In the future, we can add additional actions that should be tracked. One that comes to mind is garbage collection. Maybe another would be Hackage index updating, if we ever care about that.

I used a integer representation of the action, but I can see an argument for a `String` field to allow for greater flexibility in the future.

Reply:
I like this approach.  No particular opinion or integer vs. string.